### PR TITLE
License information update

### PR DESCRIPTION
**Pull request by sushanthakumar**
**PR Description:**
This PR is to update the license information of this project along with its dependencies
License file information in README file is updated accordingly
Project license is updated from MIT to Apache Version 2.0 considering the project use cases and deployment scenarios.

### CODE REVIEW

**Comment by sushanthakumar** [comment]
@swaminathanvasanth @abhi4578 Please have a look

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,229 @@
...
+[If you notice, any link broken or suggestions, please report through an issue to the corresponding repository]
+<Module / Repository >  <License Information>
+Elastic Search Server   Server Side Public License VERSION 1, ELASTIC-LICENSE-2.0,
+                        Apache License Version 2.0
+Elastic Search Client   Apache License Version 2.0
+Postgres                PostgreSQL Licence
+Postgres Client         BSD 2-Clause "Simplified" License
+RabbitMQ Client         Apache License Version 2.0
+Zookeeper               Hazelcast Community License Agreement, Version 1.0
+
```
Comment (swaminathanvasanth) [inline LICENSE:226]: Need to add Immudb <https://github.com/codenotary/immudb/blob/master/LICENSE>

File: LICENSE [hunk 2]
```diff
@@ -1,21 +1,229 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (swaminathanvasanth) [inline LICENSE:190]: Copyright [2024] [IUDX Programme Unit, Society of Innovation and Development, Indian Institute of Science, Bangalore]

**Review by swaminathanvasanth — CHANGES_REQUESTED** [review]
Add the following: 
L190: Copyright [2024] [IUDX Programme Unit, Society of Innovation and Development, Indian Institute of Science, Bangalore]
L226: Add Immudb <https://github.com/codenotary/immudb/blob/master/LICENSE>

**Comment by sushanthakumar** [comment]
Ok , will use this

**Review by swaminathanvasanth — CHANGES_REQUESTED** [review]
dep-licenses: 
L4,L5 are the same

Have updated the excel to include groupID:artifactID for all the Dependant Module/Repository column. Can we update the column as per the reference sheet?

**Comment by sushanthakumar** [comment]
@swaminathanvasanth @abhi4578 Dependency licenses file updated with all the licenses related to servers and client packages, pls review

**Review by swaminathanvasanth — APPROVED** [review]
LGTM

**Pull request merged**